Library/Homebrew/cask/artifact/moved.rb

```diff
@@ -125,13 +151,25 @@ def move_back(skip: false, force: false, command: nil, **options)
         delete(target, force: force, command: command, **options)
       end
 
-      def delete(target, force: false, command: nil, **_)
+      def delete(target, force: false, successor: nil, command: nil, **_)
         ohai "Removing #{self.class.english_name} '#{target}'"
         raise CaskError, "Cannot remove undeletable #{self.class.english_name}." if MacOS.undeletable?(target)
 
         return unless Utils.path_occupied?(target)
 
-        if target.parent.writable? && !force
+        if !successor.nil? && target.directory? && successor.artifacts.any? do |a|
+             a.class.dsl_key == self.class.dsl_key && a.target == self.target
```

Comment:
```suggestion
             a.class == self.class && a.target == self.target
```

Reply:
Updated, with one caveats from the style checker. It doesn't like an equality test on the classes, so I used a two-way `instance_of?` check that has the same behavior.